plugins/cost-insights/src/utils/assert.ts

```diff
@@ -50,3 +50,9 @@ export function findAlways<T>(
 ): T {
   return assertAlways(collection.find(callback));
 }
+
+export function findFirstKey<T>(
+  record: Record<string, T> | undefined,
+): string | undefined {
+  return Object.keys(record ?? {}).find(_ => true);
+}
```

Comment:
FYI, enumeration order of properties is a bit of a dicey subject and doesn't necessarily follow insertion order in some contexts.

https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order/38218582#38218582

Reply:
The usage of this function expects to only find one key, so maybe `findAnyKey` would be a more appropriate name here. The point stands in the case of dialog tab order though, I use `Object.keys` similarly there. I'm not sure a good way around it though, the `entities` keys will likely be a cast object at runtime 🤔  It sounds at least like string keys may (coincidentally due to browser whims) preserve order.

Reply:
Could enforce order by sorting lexicographically if we want to be super safe

Reply:
Eh, doesn't bother me 🤷‍♂️